### PR TITLE
feat(hud): expose the PT4 action enums as F/C/R HUD stats

### PR DESCRIPTION
**Pull request by jejellyroll-fr**
Follow-up to #267, which stored the twenty `enum_*_action` fields in `HandsPlayers` but left them without a reader. This adds the read side: 51 HUD stats, the full call / raise / fold split of all seventeen spots the enums record.

## Why a counting layer was needed

The HUD does not read `HandsPlayers`. `get_stats_from_hand_aggregated` sums `HudCache`, and a `CHAR(1)` response cannot be summed. PT4 solves the same problem at query time —

```sql
sum(if[cash_hand_player_statistics.enum_p_3bet_action = 'R', 1, 0])
```

— but FPDB's cache is additive, so the counting has to happen once at import instead.

`action_enum_stats.py` is the single source of truth: the seventeen situations, and a `derive_counters()` that turns each response char into three counters — **faced**, **called**, **raised**. Column names are PT4's own (`cnt_p_raise_3bet`), so a PT4 stat definition naming them maps straight onto FPDB.

The fold leg is deliberately **not** stored: a faced situation resolves to exactly one of F/C/R, so `folded = faced - called - raised`. Storing it would add seventeen columns that can never disagree with the other three.

## Where the counters live, and why

`HUDCACHE_EXTRA_KEYS`, not `CACHE_KEYS`. `storeHudCache` reads that list with `.get(k, 0)`, so the counters need **no `HandsPlayers` column of their own** — the response char is already there, and a second copy in the hand table would only be one more thing to keep in step.

`ensure_hudcache_columns` already ALTERs `HUDCACHE_EXTRA_KEYS` in as `INT DEFAULT 0`, so an existing database picks the 51 columns up on its next connect and **`DB_VERSION` stays at 224** — no "Please recreate tables". Counters fill from the next import onward; older hands keep their enum char in `HandsPlayers`, and `rebuild_cache` recounts them from there (see below).

## What is actually new

The fold legs overlap the existing `foldToStreetNCB*` / `street0_FoldTo3B*` / `street0_FoldToSqueeze*` pairs and do not replace them. Those count their own opportunity flags, computed separately from the enum's "situation arose", so the two answer slightly different questions and can differ by a hand or two.

What did not exist before:

- **call and raise frequencies** for any of these spots — only the fold side was ever exposed
- **any donk-bet stat at all**; `flg_f_donk_def_opp` and friends were opportunity flags with no stat reading them
- three legs of one spot sharing a denominator, so they **always add to 100%**, which the older chance/done pairs cannot do

Examples: `raise_vs_flop_cbet` (PT4's "Raise CBet"), `call_vs_preflop_squeeze`, `fold_vs_turn_donk`, `raise_float_turn`.

## Surface touched

| file | change |
|---|---|
| `action_enum_stats.py` | new — situations, counter names, derivation |
| `stats_action_enums.py` | new — 51 stat functions + the derived fold ratio |
| `DerivedStats.py` | derive the counters once the enums are computed |
| `database_caches.py` | `HUDCACHE_EXTRA_KEYS` += the 51 |
| `sql_queries_hud_cache_write.py` | insert columns/placeholders, update SET clauses |
| `sql_schema_hud_cache.py` | DDL for MySQL, PostgreSQL and SQLite |
| `sql_queries_hud_aggregated_stats.py`, `sql_queries_hud_current_stats.py` | select the counters as `stat_dict` keys |
| `Stats.py` | 51 re-exports so `STATLIST` picks them up |
| `.github/workflows/ci.yml` | both new modules added to the mypy list |

## Verification

- full suite: **8164 passed**, 16 skipped, 2 failed
  - the 2 failures are the pre-existing iPoker golden snapshots, unchanged since before #266
- `ruff check fpdb_3_legacy fpdb test tests tools`: clean
- `mypy` on the new and changed modules: No issues found
- the widened `get_stats_from_hand_aggregated` is executed against a real SQLite schema in the tests, not just string-matched — all 51 counters come back as `stat_dict` keys

`test_action_enum_hud_stats.py` (26 tests) covers derivation, the counters agreeing with the enums on a parsed hand, and the rendering edge cases. Most of it is structural, because the failure mode here is three lists drifting apart. One worth calling out: `test_insert_column_order_matches_the_key_lists` compares the insert's column list against `CACHE_KEYS + HUDCACHE_EXTRA_KEYS` **in order**. `storeHudCache` binds positionally and the existing sync test only counts columns, so two transposed names would file every raise under calls and still pass today.

## Every path that reads the counters

Three reads, not one, and the first version of this branch only covered one of them — both gaps were caught in review:

| path | when | how the counters arrive |
|---|---|---|
| `get_stats_from_hand_aggregated` | normal HUD | `sum(hc.…)` out of HudCache |
| `get_stats_from_hand_session` | `stat_range == "S"` | `CASE` over `hp2.enum_*`, summed in Python |
| `rebuildCache` | `rebuild_cache(table="HudCache")` | `sum(CASE WHEN hp.enum_x = 'R' …)` re-counted from HandsPlayers |

The rebuild one matters twice over: `rebuild_cache` **clears** HudCache before refilling it, so without naming the counters a rebuild would have dropped everything accumulated since the upgrade *and* left the stats empty. Deriving them from the stored response char also means an existing history that already has the enums (imported since #267) gets its counters by rebuilding, with no re-import.

The rebuild template's column list is shared with `CardsCache` and `PositionsCache`, which have no such columns, so it grew two placeholders that only `_statscache_hudcache` fills and `replace_statscache` strips everywhere else.

### Pre-existing rebuild gap, not fixed here

While wiring that up: `rebuildCache` names **115 of 253** `CACHE_KEYS` and, before this branch, **none** of `HUDCACHE_EXTRA_KEYS`. `flg_faced_allin`, the probe and delayed-c-bet pairs, and roughly 138 other columns are still zeroed by a rebuild today. That is long-standing and out of scope here, but the placeholder pair this PR adds is the hook a later fix can reuse.

## Noted in passing, not fixed here

`get_stats_from_hand` (the non-aggregated HUD read) does not parse under SQLite — `near "sum": syntax error`. I verified this on `5eaa4380`, before any change in this PR. Out of scope here, but it means that query is either unused or MySQL/PostgreSQL-only, and it is worth a look.

### CODE REVIEW

File: fpdb_3_legacy/database_caches.py [hunk 1]
```diff
@@ -297,6 +298,11 @@
     "street2DelayedCBDone",
     "street2ProbeChance",
     "street2ProbeDone",
+    # PT4 action-enum counters. They live here rather than in CACHE_KEYS because
+    # HandsPlayers already stores the response char they are counted from; only
+    # the HUD needs them pre-summed. storeHudCache reads them with .get(k, 0),
+    # so no HandsPlayers column is involved.
+    *ACTION_ENUM_CACHE_KEYS,
```
[the remark this answers is not in the document]
Comment (jejellyroll-fr) [reply fpdb_3_legacy/database_caches.py:305]: Legitimate and fixed in efaf30cc — thank you, this was the sharper of the two.

Two things were true at once. `rebuildCache` did not name the counters, so a rebuild left them at zero; worse, `rebuild_cache` **clears** HudCache first, so it also dropped whatever had accumulated since the upgrade. The PR description claimed a rebuild would recover them, which was simply wrong. Corrected.

The rebuild now derives them from the stored response char, which is still in HandsPlayers:

    sum(CASE WHEN hp.enum_p_3bet_action = 'R' THEN 1 ELSE 0 END)

That template's column list is shared with CardsCache and PositionsCache, which have no such columns, so it grew two placeholders that only `_statscache_hudcache` fills and `replace_statscache` strips elsewhere. Three tests drive the real `_rebuild_ring_cache` against a SQLite schema with a seeded hand and assert the counters come back right, rather than matching query text.

One correction to the framing: this is not specific to the new columns. `rebuildCache` names 115 of 253 `CACHE_KEYS` and, before this branch, none of `HUDCACHE_EXTRA_KEYS` — `flg_faced_allin`, the probe and delayed-c-bet pairs and ~138 other columns are still zeroed by a rebuild today. I fixed it for these 51 and left the rest; the placeholder pair added here is the hook that later fix can reuse.

File: fpdb_3_legacy/sql_queries_hud_current_stats.py [hunk 2]
```diff
@@ -71,6 +71,58 @@ def hud_current_stats_queries() -> dict[str, str]:
                 sum(hc.flg_blind_k)                 AS straddle_done,
                 sum(hc.flg_faced_allin)             AS faced_allin,
                 sum(hc.flg_fold_to_allin)           AS fold_allin,
+                /* PT4 action-enum counters (action_enum_stats.py). */
+                sum(hc.cnt_p_face_3bet)             AS cnt_p_face_3bet,
```
[the remark this answers is not in the document]
Comment (jejellyroll-fr) [reply fpdb_3_legacy/sql_queries_hud_current_stats.py:75]: Legitimate and fixed in efaf30cc.

Confirmed the mechanism before fixing: `get_stats_from_hand` hands any player on `stat_range == "S"` (or hero on `h_stat_range`) to `get_stats_from_hand_session`, whose three backend queries read HandsPlayers through `hp2` and aggregate in Python rather than summing HudCache. None of the 51 counters existed there, so session mode rendered no-data for all of them.

All three variants now carry the same counters as CASE expressions over `hp2.enum_*`, under identical aliases:

    CASE WHEN hp2.enum_f_cbet_action = 'R' THEN 1 ELSE 0 END AS cnt_f_raise_cbet

One detail that made this work without further changes: that path lowercases column names before summing them into `stat_dict`, and the counter names are already lowercase. There is a test pinning that, so a future camelCase counter cannot silently break session mode.

Covered for all three backends, and those three tests fail without the commit.

**Comment by jejellyroll-fr** [comment]
@codex review
